app/src/main/java/com/sun/premierleague/base/BaseFragment.kt

```diff
@@ -0,0 +1,44 @@
+package com.sun.premierleague.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.sun.premierleague.utils.showToast
+
+abstract class BaseFragment : Fragment(), BaseView {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initPresenter()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(configView(), container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initData(view)
+    }
+
+    protected abstract fun configView(): Int
+
+    protected abstract fun initPresenter()
+
+    protected abstract fun initData(v: View)
```

Comment:
Tại sao initData lại có cả view trong đây nhỉ? 
1. Tách biệt View và data bằng bằng 2 hàm initView(), initData()
2. Bản thân không cần truyền view vào vì view đã được lưu trong biến toàn cục view rồi 
Viết trong onViewCreated thì View đã được tạo rồi nên không cần view nữa